### PR TITLE
feat(AwsRedisInstance): add API to specify parameters

### PR DESCRIPTION
**Pull request by dushanpantic**
**Description**

Changes proposed in this pull request:

- Ability for user to define redis parameters
- If parameter is not defined, it will fall back to default value

out of scope:
- validation

**Related issue(s)**

### CODE REVIEW

File: pkg/kcp/provider/aws/redisinstance/client/elastiCacheClient.go [hunk 1]
```diff
@@ -101,6 +110,120 @@ func (c *client) DeleteElastiCacheSubnetGroup(ctx context.Context, name string)
...
+	return nil
+}
+
+func (c *client) DescribeElastiCacheParameters(ctx context.Context, groupName string) ([]elasticacheTypes.Parameter, error) {
+	result := []elasticacheTypes.Parameter{}
+	var marker *string = nil
+	for {
+		out, err := c.elastiCacheSvc.DescribeCacheParameters(ctx, &elasticache.DescribeCacheParametersInput{
+			CacheParameterGroupName: ptr.To(groupName),
+			Marker:                  marker,
```
Comment (ijovovic) [inline pkg/kcp/provider/aws/redisinstance/client/elastiCacheClient.go:161]: If the marker is returned as non nil in `out` var, will it actually be assigned to `marker` value in next request? If not, we might get into endless loop.

File: pkg/kcp/provider/aws/redisinstance/util.go [hunk 2]
```diff
@@ -1,11 +1,98 @@
...
 func GetAwsElastiCacheSubnetGroupName(name string) string {
 	return fmt.Sprintf("cm-%s", name)
 }
 
+func GetAwsElastiCacheParameterGroupName(name string) string {
+	return fmt.Sprintf("cm-%s", name)
+}
+
+func GetAwsElastiCacheParameterGroupFamily(engineVersion string) string {
+	if strings.Contains(engineVersion, "2.6") {
```
Comment (ijovovic) [inline pkg/kcp/provider/aws/redisinstance/util.go:20]: In the case of engine 3.2.6 (https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/ParameterGroups.Redis.html#ParameterGroups.Redis.3-2-6) we might get the wrong param group family.

Comment (dushanpantic) [reply pkg/kcp/provider/aws/redisinstance/util.go]: change to check prefix

File: pkg/kcp/provider/aws/redisinstance/client/elastiCacheClient.go (hunk 1 above)
Comment (dushanpantic) [reply pkg/kcp/provider/aws/redisinstance/client/elastiCacheClient.go:161]: Line 170 would stop endless loop
```
                if out.Marker == nil {
			break
		}
```
But you are right, marker was not being reassigned, nice catch.
I will add`marker = out.Marker` at end of the loop

**Review by ijovovic — APPROVED** [review]
LGTM

**Pull request merged**